homeassistant/components/ipp/config_flow.py

```diff
@@ -0,0 +1,160 @@
+"""Config flow to configure the IPP integration."""
+import logging
+from typing import Any, Dict, Optional
+
+from pyipp import IPP, IPPConnectionError, IPPConnectionUpgradeRequired
+import voluptuous as vol
+
+from homeassistant.config_entries import CONN_CLASS_LOCAL_POLL, ConfigFlow
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .const import CONF_BASE_PATH, CONF_UUID
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: HomeAssistantType, data: dict) -> Dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    session = async_get_clientsession(hass)
+    ipp = IPP(
+        host=data[CONF_HOST],
+        port=data[CONF_PORT],
+        base_path=data[CONF_BASE_PATH],
+        tls=data[CONF_SSL],
+        verify_ssl=data[CONF_VERIFY_SSL],
+        session=session,
+    )
+
+    printer = await ipp.printer()
+
+    return {CONF_UUID: printer.info.uuid}
+
+
+class IPPFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle an IPP config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_LOCAL_POLL
+
+    discovery_info = {}
```

Comment:
By defining it in the class definition, the dictionary is shared across all instances of this class. If 2 flows were to be set up at the same time, it will break! Define it inside `__init__` instead.
```suggestion
    def __init__(self):
        discovery_info = {}
```